nautobot_circuit_maintenance/handle_notifications/sources.py

```diff
@@ -287,26 +296,34 @@ def fetch_email(self, logger: Job, msg_id: bytes, since: Optional[int]) -> Optio
             logger.log_failure(message=f"Not possible to determine the email sender: {email_message['From']}")
             return None
 
-        provider_data_type, provider_type, error_message = self.extract_provider_data_type(email_source)
-        if not provider_data_type:
+        provider_data_types, provider_type, error_message = self.extract_provider_data_types(email_source)
+        if not provider_data_types:
             logger.log_warning(message=error_message)
             return
 
-        for part in email_message.walk():
-            if part.get_content_type() == provider_data_type:
-                raw_payload = part.get_payload()
-                break
+        raw_payloads = []
+        for provider_data_type in provider_data_types:
+            for part in email_message.walk():
+                if part.get_content_type() == provider_data_type:
+                    raw_payloads.append(part.get_payload())
+                    break
+            else:
+                logger.log_debug(
+                    message=f"Payload type {provider_data_type} not found in email payload.",
+                )
+                continue
+            break
         else:
             logger.log_warning(
-                message=f"Payload type {provider_data_type} not found in email payload.",
+                message=f"Payload types {', '.join(provider_data_types)} not found in email payload.",
             )
```

Comment:
I'm actually not sure this is working as it's supposed to, do you want to only get the first payload that matches the first provider_data_type? Could definitely do with some comments about the intended flow, and consider again breaking this into a helper function to simplify the flow control logic.

Reply:
good catch!